gobblin-utility/src/main/java/gobblin/util/filesystem/AbstractPathComparator.java

```diff
@@ -0,0 +1,35 @@
+package gobblin.util.filesystem;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import org.apache.hadoop.fs.Path;
+
+
+abstract class AbstractPathComparator implements Comparator<Path> {
```

Comment:
You don't need this anymore when not using case sensitivity. `Path` is comparable.